src/components/FeatureInformation/FeatureInformation.js

```diff
@@ -1,17 +1,41 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import PropTypes from "prop-types";
 import { useTranslation } from "react-i18next";
 import { MdClose } from "react-icons/md";
 import { IoIosArrowRoundBack, IoIosArrowRoundForward } from "react-icons/io";
 import { Link, IconButton } from "@mui/material";
+import { unByKey } from "ol/Observable";
 import { setFeatureInfo } from "../../model/app/actions";
 import useIndexedFeatureInfo from "../../utils/useIndexedFeatureInfo";
 import useHighlightLayer from "../../utils/useHighlightLayer";
 import getPopupComponent from "../../utils/getPopupComponent";
 import "./FeatureInformation.scss";
 
+const useUpdateFeatureInfoOnLayerDeactivate = (layers, featureInfo) => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    const listeners = [];
+    layers.forEach((l) => {
+      const listener = l?.on("change:visible", (evt) => {
+        const layer = evt.target;
+        if (!layer.visible) {
+          layer.select();
+          layer.highlight();
```

Comment:
are select and highlight functions available for all layers ?    use layer?.select to be sure

Reply:
ok